smart_open/smart_open_lib.py

```diff
@@ -91,11 +91,11 @@ def smart_open(uri, mode="rb"):
 
     if mode in ('r', 'rb'):
         if parsed_uri.scheme in ("s3", "s3n"):
-            return S3OpenRead(parsed_uri)
+            return S3OpenRead(parsed_uri, **kw)
         elif parsed_uri.scheme in ("hdfs", ):
-            return HdfsOpenRead(parsed_uri)
+            return HdfsOpenRead(parsed_uri, **kw)
         elif parsed_uri.scheme in ("webhdfs", ):
-            return WebHdfsOpenRead(parsed_uri)
+            return WebHdfsOpenRead(parsed_uri, **kw)
```

Comment:
This extra constructor confused my merge, so I had to tell it specifically to use the **kw argument for the new ctor.